airflow-core/docs/installation/upgrading_to_airflow3.rst

```diff
@@ -0,0 +1,130 @@
+Upgrading to Airflow 3
+=======================
+
+Apache Airflow 3 is a major release. The purpose of this document is to guide users in their upgrade from Airflow 2.x to Airflow 3.
+
+Step 1: Take care of prerequisites
+----------------------------------
+
+- Make sure that you are on Airflow 2.7 or later.
+- Make sure that your Python version is in the supported list. Airflow 3 supports the following Python versions: Python 3.9, 3.10, 3.11 and 3.12.
+- Ensure that you are not using SubDAGs. These were deprecated in Airflow 2.0 and unavailable in Airflow 3.
+- For a complete list of breaking changes, which you should note before the upgrade, please check the breaking changes section below.
+
+Step 2: Clean and back up your existing Airflow Instance
+---------------------------------------------------------
+
+- It is highly recommended to make a backup of your Airflow instance specifically including your Airflow metadata DB before starting the migration process.
+- If you do not have a "hot backup" capability for your DB, you should do it after shutting down your Airflow instances, so that the backup of your database will be consistent.
+- If you did not make a backup and your migration fails, you might end-up in a half-migrated state and restoring DB from backup and repeating the migration might be the only easy way out. This can for example be caused by a broken network connection between your CLI and the database while the migration happens, so taking a backup is an important precaution to avoid problems like this.
+- A long running Airflow instance can accumulate a certain amount of silt, in the form of old database entries, which are no longer required. This is typically in the form of old XCom data which is no longer required, and so on. As part of the Airflow 3 upgrade process, there will be schema changes. Based on the size of the Airflow meta-database this can be somewhat time consuming. For a faster, safer migration, we recommend that you clean up your Airflow meta-database before the upgrade.
```

Comment:
```suggestion
- A long running Airflow instance can accumulate a certain amount of silt, in the form of old database entries, which are no longer required. This is typically in the form of old XCom data which is no longer required, and so on. As part of the Airflow 3 upgrade process, there will be schema changes. Based on the size of the Airflow meta-database this can be somewhat time consuming. For a faster, safer migration, we recommend that you clean up your Airflow meta-database before the upgrade. You can use `airflow db clean` command for that.
```